jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -280,23 +279,12 @@ def get_link_type(self, data_label):
         if len(self.session.application.data_collection) == 0:
             raise ValueError('No reference data for link look-up')
 
-        # the original links were created against data_collection[0], not necessarily
+        # TODO: Brett Morris might want to look at this for
+        # https://github.com/spacetelescope/jdaviz/pull/2179
+        #     ref_label = self.state.reference_data ???
+        #
+        # The original links were created against data_collection[0], not necessarily
         # against the current viewer reference_data
         ref_label = self.session.application.data_collection[0].label
```

Comment:
This should work on this branch _and_ in https://github.com/spacetelescope/jdaviz/pull/2179:

```suggestion
        ref_label = self.state.reference_data.label
```

Reply:
But this reverts change from @kecnry in #1908

Reply:
This is going to require some thought.... @bmorris3 in #2179 are links always updated wrt reference data?  I think until that is merged, we might need to leave this at `data_collection[0]`, but would love to be convinced otherwise.

Reply:
> in https://github.com/spacetelescope/jdaviz/pull/2179 are links always updated wrt reference data?

Links should be updated only when necessary in #2179.

I'm ok with holding off on the change and cleaning up if/when it's necessary in 2179. Feel free to mark as resolved.